### ISSUE TITLE
Rule proposal: no-dupe-characters-character-class

### ISSUE THREAD

**Issue by ota-meshi** [eslint/eslint#13341]
**Please describe what the rule should do:**

This rule reports if there are duplicate characters in the RegExp character class.

Finding duplicate characters may help find other problems. I think this problem often appears when you use the escape incorrectly.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
var re = /[\\(\\)]/
//         ^^ ^^        "\\" are duplicated 
var re = /[a-z\\s]/
//         ^^^  ^       "s" are duplicated 
```

**Why should this rule be included in ESLint (instead of a plugin)?**

I would like to have rules in ESLint because of the problems that can occur with general JavaScript.

**Are you willing to submit a pull request to implement this rule?**

If you accept this rule, I will pull request this rule.

**Comment by ota-meshi** [comment]
I readed the [Changes to our Rules Policies](https://eslint.org/blog/2020/05/changes-to-rules-policies).
I understand that you probably won't accept this rule, if you know of a plugin repository that would suggest this rule please let me know.

**Comment by nzakas** [comment]
Thanks for the suggestion. As you mentioned, this doesn't really qualify for consideration as a new rule in the core. I was trying to consider if this would catch any errors, but it seems like this is really more of a preference. While I can't recommend any specific plugin for this rule, you can always create your own plugin to distribute it.

Thanks for your understanding.

**Comment by ota-meshi** [comment]
I'm creating this rule with plugin.
https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/docs/rules/no-dupe-characters-character-class.md
